### PR TITLE
Include minidump-stackwalk errors in minidump logs

### PR DESCRIPTION
**Pull request by tysmith**
This makes minidump reporting function more like sanitizers.

### CODE REVIEW

File: src/ffpuppet/minidump_parser.py [hunk 1]
```diff
@@ -121,37 +134,63 @@ def format_output(
...
+            success = True
+        except (CalledProcessError, TimeoutExpired):
             LOG.error("Failed to process: %s", src)
-            raise MinidumpStackwalkFailure(b"minidump-stackwalk failed") from None
-        except TimeoutExpired:
-            LOG.error("Failed to process: %s", src)
-            raise MinidumpStackwalkFailure("minidump-stackwalk hung") from None
-        return out_json
+            success = False
+        return success, out_json
```
Comment (jschwartzentruber) [inline src/ffpuppet/minidump_parser.py:164]: Why not just return `Optional[Path]` and return `None` on failure?

File: src/ffpuppet/minidump_parser.py [hunk 2]
```diff
@@ -121,37 +134,63 @@ def format_output(
...
         cmd.append(str(src))
+        # get unique file name
+        with NamedTemporaryFile(dir=self._storage, prefix="mdsw_out_") as ofp:
+            out_json = Path(ofp.name)
         LOG.debug("running %r", " ".join(cmd))
         try:
-            run(cmd, check=True, stdout=DEVNULL, timeout=60)
-        except CalledProcessError:
+            with out_json.open("wb") as out_fp:
+                run(cmd, check=True, stderr=out_fp, stdout=out_fp, timeout=60)
```
Comment (jschwartzentruber) [inline src/ffpuppet/minidump_parser.py:159]: Redirecting stderr means any warning message is going to corrupt the json output?

File: src/ffpuppet/minidump_parser.py (hunk 1 above)
Comment (jschwartzentruber) [reply src/ffpuppet/minidump_parser.py:164]: I see, you're returning JSON or an error log...

Comment (jschwartzentruber) [reply src/ffpuppet/minidump_parser.py:164]: I think stdout and stderr need to be separated, and you use one or the other depending on if mdsw returned success. Not returning an error log and calling it json. I would also output stderr on LOG.debug in any case if it's non-empty, or we may be hiding warnings.

**Review by jschwartzentruber — CHANGES_REQUESTED** [review]

File: src/ffpuppet/minidump_parser.py (hunk 1 above)
Comment (tysmith) [reply src/ffpuppet/minidump_parser.py]: I tried to clean things up by breaking it up but I just ended up making a mess. This should be resolved now.

**Review by jschwartzentruber — APPROVED** [review]

**Pull request merged**